front/src/components/oauth2/actions.js

```diff
@@ -0,0 +1,112 @@
+import { RequestStatus } from '../../utils/consts';
+
+const actions = store => ({
+  updateIntegrationName(state, e) {
+    store.setState({
+      integrationName: e
+    });
+  },
+  updateClientId(state, e) {
+    store.setState({
+      clientId: e.target.value
+    });
+  },
+  updateSecretId(state, e) {
+    store.setState({
+      secretId: e.target.value
+    });
+  },
+  async getCurrentConfig(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+
+    const returnServiceId = await state.httpClient.get(`/api/v1/service/${state.integrationName}/getServiceId`);
+
+    const returnGetConfig = await state.httpClient.get('/api/v1/service/oauth2/getCurrentConfig', {
+      integrationName: state.integrationName,
+      serviceId: returnServiceId.result.serviceId
+    });
+
+    if (returnGetConfig) {
+      // Case of config found
+      store.setState({
+        clientIdInDb: returnGetConfig.clientId,
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
+      });
+    } else {
+      store.setState({
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
+      });
+    }
+  },
+  async startConnect(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+    try {
+      if (!state.secretId || !state.clientId) {
+        store.setState({
+          oauth2GetStatus: RequestStatus.Error,
+          oauth2ErrorMsg: 'errorEmptyConfig'
+        });
+        return;
+      }
+
+      const returnSaveVar = await state.httpClient.post(`/api/v1/service/${state.integrationName}/savevar`, {
```

Comment:
What `savevar` means?

Reply:
it's to save variable in t_variable table with serviceId column